### ISSUE TITLE
What is the type of the exported configuration function

### ISSUE THREAD

**Issue by OnkelTem** [webpack/webpack#14508]
Now when webpack is well typed and provides 12000+ lines of typescript code, it would be great to have a type for the main configuration function - the one which is exported.

Here is a question I asked a few days ago on StackOverflow: https://stackoverflow.com/questions/69584268/what-is-the-type-of-the-webpack-config-function-when-it-comes-to-typescript

Let me repeat it here:

Currently when one exports an object from the config-file, the `Configuration` type can be used:

```ts
const config: Configuration = {
  mode: 'production',
  // ...
}
```

But what is the type for exporting the function? E.g. 

```ts
export default (...): ... => {
  return {
    mode: 'production'
    // ...
  }
}
```

Surprisingly, but I couldn't the proper type among a bunch of the types provided for webpack in https://github.com/webpack/webpack/blob/main/types.d.ts

If you followed the SO line above, you'd see the answer from Igor: https://stackoverflow.com/a/69608855/1223483
Basically he makes this job: implementing _missed_ types. 

But how could they be missed?

**Comment by alexander-akait** [comment]
Duplicate https://github.com/webpack/webpack/issues/11630, please use https://github.com/webpack/webpack/issues/11630#issuecomment-838510078
